src/platform/ESP32/PlatformManagerImpl.cpp

```diff
@@ -118,6 +118,24 @@ CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
     return chip::DeviceLayer::Internal::ESP32Utils::MapError(err);
 }
 
+CHIP_ERROR PlatformManagerImpl::_GetCurrentHeapFree(uint64_t & currentHeapFree)
+{
+    currentHeapFree = heap_caps_get_free_size(MALLOC_CAP_8BIT);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetCurrentHeapUsed(uint64_t & currentHeapUsed)
+{
+    currentHeapUsed = heap_caps_get_total_size(MALLOC_CAP_8BIT) - heap_caps_get_free_size(MALLOC_CAP_8BIT);
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR PlatformManagerImpl::_GetCurrentHeapHighWatermark(uint64_t & currentHeapHighWatermark)
+{
+    currentHeapHighWatermark = heap_caps_get_free_size(MALLOC_CAP_8BIT);
```

Comment:
This should be the maximum amount of heap memory in bytes, that has been used by the node.
So, I think, this should be: total heap size - minimum heap ever. There's an API for the minimum heap ever: `heap_caps_get_minimum_free_size(uint32_t capabilities)` / `esp_get_minimum_free_heap_size`

Reply:
Done